### PR TITLE
feat(trivy): Bump to support v0.56.1

### PR DESCRIPTION
**Pull request by simar7**

### CODE REVIEW

**Comment by simar7** [comment]
@nikpivkin any idea why the tests are red in the CI? Locally I don't see any failures.

```shell
 BATS_LIB_PATH=/opt/homebrew/lib TRIVY_DISABLE_VEX_NOTICE=true bats -r -T .
./test/test.bats
 ✓ trivy repo with securityCheck secret only [1000]
 ✓ trivy image [1000]
 ✓ trivy config sarif report [1000]
 ✓ trivy config [1000]
 ✓ trivy rootfs [1000]
 ✓ trivy fs [1000]
 ✓ trivy image with trivyIgnores option [1000]
 ✓ trivy image with sbom output [1000]
 ✓ trivy image with trivy.yaml config [1000]
 ✓ trivy image with custom docker-host [1000]
 ✓ trivy config with terraform variables [1000]

11 tests, 0 failures in 12 seconds
```

**Comment by nikpivkin** [comment]
@simar7 The tests use the `trivy-checks` bundle snapshot stored in this repository, but running the tests locally uses checks from the Trivy cache, so the test result is different. I will open a separate PR with an update to the Makefile to make the local test run consistent with CI.

This is the difference in local run on the main branch:
```bash
✗ trivy config sarif report [607]
   (from function `assert_files_equal' in file /opt/homebrew/lib/bats-file/src/file.bash, line 266,
    from function `compare_files' in file ./test/test.bats, line 54,
    in test file ./test/test.bats, line 72)
     `compare_files config-sarif.sarif ./test/data/config-sarif-report/report.sarif' failed
   Building SARIF report with options:  ./test/data/config-sarif-report/main.tf
   3c3
   <   "$schema": "https://raw.githubusercontent.com/oasis-tcs/sarif-spec/main/sarif-2.1/schema/sarif-schema-2.1.0.json",
   ---
   >   "$schema": "https://raw.githubusercontent.com/oasis-tcs/sarif-spec/master/Schemata/sarif-schema-2.1.0.json",
   3c3
   <   "$schema": "https://raw.githubusercontent.com/oasis-tcs/sarif-spec/main/sarif-2.1/schema/sarif-schema-2.1.0.json",
   ---
   >   "$schema": "https://raw.githubusercontent.com/oasis-tcs/sarif-spec/master/Schemata/sarif-schema-2.1.0.json",
   
   -- files are not the same --
   path : config-sarif.sarif
   path : ./test/data/config-sarif-report/report.sarif
   --
```

**Comment by simar7** [comment]
@nikpivkin I've updated the PR to Trivy v0.56.0, when it is released we can merge this one in.

File: Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM ghcr.io/aquasecurity/trivy:0.53.0
+FROM ghcr.io/aquasecurity/trivy:0.55.2
```
Comment (BertelBB) [inline Dockerfile:1]:
```suggestion
FROM ghcr.io/aquasecurity/trivy:0.56.0
```

**Comment by simar7** [comment]
@nikpivkin could you take another look?

File: test/data/config-sarif-report/report.sarif [hunk 2]
```diff
@@ -9,87 +9,6 @@
           "informationUri": "https://github.com/aquasecurity/trivy",
           "name": "Trivy",
           "rules": [
-            {
-              "id": "AVD-AWS-0086",
-              "name": "Misconfiguration",
-              "shortDescription": {
-                "text": "S3 Access block should block public ACL"
-              },
```
Comment (simar7) [inline test/data/config-sarif-report/report.sarif:17]: does this make sense to you @nikpivkin? I understand that the Go rule was deprecated but should the Rego rule in it's place still pick this up?

**Comment by nikpivkin** [comment]
@simar7 We need to update the checks [bundle](https://github.com/aquasecurity/trivy-action/pkgs/container/trivy-checks-act). Can you run [this](https://github.com/aquasecurity/trivy-action/actions/workflows/sync-trivy-checks.yaml) workflow? I don't have permission to do that.

**Comment by nikpivkin** [comment]
@simar7 https://github.com/aquasecurity/trivy-action/pull/398

**Comment by nikpivkin** [comment]
The tests use checks bundle from [here](https://github.com/aquasecurity/trivy-action/pkgs/container/trivy-checks-act) and it is not updated

**Comment by eliflores** [comment]
Refs https://github.com/aquasecurity/trivy-action/issues/400

**Pull request merged**